File: update_db.js
const fs = require('fs');

const sql = fs.readFileSync('Database_PostgreLatest.sql', 'utf-8');
const lines = sql.split('\n');

let newLines = [];
let inCards = false;
let inSessions = false;
let sessionCardIdx = 1;
let cardStatuses = {};

for (let i = 1; i <= 150; i++) {
    cardStatuses[String(i).padStart(3, '0')] = 'ACTIVE';
}
cardStatuses['LOST-01'] = 'LOST';
cardStatuses['LOST-02'] = 'LOST';

for (let i = 0; i < lines.length; i++) {
    let line = lines[i];

    if (line.includes('INSERT INTO ParkingCards')) {
        inCards = true;
        newLines.push(line);
        continue;
    }

    if (inCards) {
        if (line.includes(';')) {
            inCards = false;
            newLines.push('__CARDS_PLACEHOLDER__');
        }
        continue;
    }

    if (line.includes('INSERT INTO ParkingSessions')) {
        inSessions = true;
        newLines.push(line);
        continue;
    }

    if (inSessions) {
        if (line.trim() === '' || line.startsWith('--')) {
            inSessions = false;
            newLines.push(line);
            continue;
        }

        const match = line.match(/^\((\d+),\s*(\d+),\s*(\d+),\s*([^,]+),\s*(.*)/);
        if (match) {
            const sess_id = match[1];
            const veh_id = match[2];
            const slot_id = match[3];
            const old_card = match[4];
            const rest = match[5];

            const cardId = String(sessionCardIdx).padStart(3, '0');
            sessionCardIdx++;

            if (rest.includes("'PARKING'") || rest.includes("'UNPAID'") || rest.includes("'VIOLATION'")) {
                cardStatuses[cardId] = 'IN_USE';
            } else if (rest.includes("'LOST_TICKET'")) {
                cardStatuses[cardId] = 'LOST';
            } else {
                // COMPLETED or others -> ACTIVE
                cardStatuses[cardId] = 'ACTIVE';
            }

            let newLine = (, , , '', ;
            newLines.push(newLine);
            
            if (rest.includes(';')) {
                inSessions = false;
            }
            continue;
        }
    }

    newLines.push(line);
}

let cardsStrList = [];
for (const [cid, status] of Object.entries(cardStatuses)) {
    cardsStrList.push(('', ''));
}
const cardsStr = cardsStrList.join(',\n') + ';';

const finalLines = newLines.map(line => line === '__CARDS_PLACEHOLDER__' ? cardsStr : line);
fs.writeFileSync('Database_PostgreLatest.sql', finalLines.join('\n'), 'utf-8');
console.log('Updated SQL. Total sessions:', sessionCardIdx - 1);

File: update_db.py
import re

with open('Database_PostgreLatest.sql', 'r', encoding='utf-8') as f:
    lines = f.readlines()

new_lines = []
in_cards = False
in_sessions = False

card_statuses = {}
for i in range(1, 151):
    card_statuses[f'{i:03d}'] = 'ACTIVE'
card_statuses['LOST-01'] = 'LOST'
card_statuses['LOST-02'] = 'LOST'

session_card_idx = 1
for i in range(len(lines)):
    line = lines[i]
    
    if 'INSERT INTO ParkingCards' in line:
        in_cards = True
        new_lines.append(line)
        continue
    
    if in_cards:
        if ';' in line:
            in_cards = False
            new_lines.append('__CARDS_PLACEHOLDER__\n')
        continue
        
    if 'INSERT INTO ParkingSessions' in line:
        in_sessions = True
        new_lines.append(line)
        continue
        
    if in_sessions:
        if line.strip() == '' or line.startswith('--'):
            in_sessions = False
            new_lines.append(line)
            continue
            
        match = re.match(r'^\((\d+),\s*(\d+),\s*(\d+),\s*([^,]+),\s*(.*)', line)
        if match:
            sess_id, veh_id, slot_id, old_card, rest = match.groups()
            
            card_id = f'{session_card_idx:03d}'
            session_card_idx += 1
            
            if "'PARKING'" in rest or "'UNPAID'" in rest or "'VIOLATION'" in rest:
                card_statuses[card_id] = 'IN_USE'
            elif "'LOST_TICKET'" in rest:
                card_statuses[card_id] = 'LOST'
            
            new_line = f"({sess_id}, {veh_id}, {slot_id}, '{card_id}', {rest}\n"
            new_lines.append(new_line)
            if ';' in rest:
                in_sessions = False
            continue
            
    new_lines.append(line)

cards_str_list = []
for cid, status in card_statuses.items():
    cards_str_list.append(f"('{cid}', '{status}')")
cards_str = ",\n".join(cards_str_list) + ";"

final_lines = []
for line in new_lines:
    if line == '__CARDS_PLACEHOLDER__\n':
        final_lines.append(cards_str + '\n')
    else:
        final_lines.append(line)

with open('Database_PostgreLatest.sql', 'w', encoding='utf-8') as f:
    f.writelines(final_lines)

print(f"Updated SQL. Next Card IDX: {session_card_idx}")
